cookbooks/bach_spark/templates/default/spark-defaults.conf.erb

```diff
@@ -0,0 +1,14 @@
+spark.driver.extraLibraryPath=/usr/hdp/current/hadoop-client/lib/native:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64
+spark.executor.extraLibraryPath=/usr/hdp/current/hadoop-client/lib/native:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64
+spark.executor.extraJavaOptions -verbose:gc -XX:+PrintHeapAtGC -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintTenuringDistribution -XX:+UseNUMA -XX:+PrintGCApplicationStoppedTime -XX:+UseCompressedOops -XX:+PrintClassHistogram -XX:+PrintGCApplicationConcurrentTime
+spark.shuffle.consolidateFiles true
+spark.shuffle.io.numConnectionsPerPeer 2
+spark.history.fs.logDirectory <%= node['spark']['hdfs_url'] %>/spark-history
+spark.eventLog.dir <%= node['spark']['hdfs_url'] %>/spark-history
+spark.eventLog.enabled true
+spark.logConf true
+spark.dynamicAllocation.enabled true
```

Comment:
Hi Ray, the thought here is that the YARN scheduler should be limiting the resources a user can consume more than us limiting it at the Spark layer which is just one of the many higher-level languages which could be in use. (E.g. for traditional Map/Reduce a job could consume a large number of containers for maps on a large input or a large number of containers for a large reduce while otherwise being well bounded in its resource demands.)